android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AndroidBillingMapper.java

```diff
@@ -198,6 +198,55 @@ public static LaunchBillingFlowResult mapBundleToHashMapGetIntent(Bundle bundle)
     return skuDetailsList;
   }
 
+  public static SkuDetails mapSingleSkuDetails(String skuType, String skuDetailsResponse) {
+    SkuDetails skuDetails =
+        new SkuDetails(skuType, "", "", "", 0, "", "", 0, "", "", 0, "", "", "");
+    if (!skuDetailsResponse.equals("")) {
+      try {
+        JSONObject jsonElement = new JSONObject(skuDetailsResponse);
+        JSONArray items = jsonElement.getJSONArray("items");
+        if (!items.isNull(0)) {
+          JSONObject obj = items.getJSONObject(0);
+
+          String sku = obj.getString("name");
+
+          JSONObject priceObj = obj.getJSONObject("price");
+
+          String price = priceObj.getJSONObject("fiat")
+              .getString("value");
+          long priceAmountMicros = getFiatAmountInMicros(priceObj.getJSONObject("fiat"));
+          String priceCurrencyCode = getFiatCurrencyCode(priceObj.getJSONObject("fiat"));
+          if (priceObj.has("base") && priceObj.getString("base")
+              .equalsIgnoreCase(APPC)) {
+            price = getAppcPrice(priceObj);
+            priceAmountMicros = getAppcAmountInMicros(priceObj);
+            priceCurrencyCode = APPC;
+          }
```

Comment:
The errors are dealt in the presenter. If there's any problem it will be catched and an error Json (default) will be created, when it reaches the presenter, the presenter checks for the Object (skuDetails) and deal with it there